internal/condition/condition.go

```diff
@@ -0,0 +1,35 @@
+package condition
+
+/*
+NOTE: To avoid cyclic dependencies, the actual struct datatype is not defined here.
+It is doubly defined - once in project_config.go and once in module_config.go.
+*/
+
+import (
+	"os"
+	"path"
+
+	"github.com/commitdev/zero/internal/config/projectconfig"
+)
+
+func Perform(cond projectconfig.Condition, mod projectconfig.Module) {
+	value, found := mod.Parameters[cond.MatchField]
+
+	// Exit if the condition isn't met.
+	if !found || value != cond.WhenValue {
+		return
+	}
+
+	// Okay, the condition was met, let's execute it.
+	switch cond.Action {
+	case "ignoreFile":
+		ignoreFile(cond.Data, mod)
+	}
+}
+
+func ignoreFile(data []string, mod projectconfig.Module) {
```

Comment:
maybe rename to `ignoreFiles`, or else it may be hard for module creators to understand `data` is a list of strings

Reply:
That's fair.

Reply:
We should be able to specify either files or directories in this list as well.